libraries/common/providers/loading/index.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { Component } from 'react';
```

Comment:
Please add a `shouldComponentUpdate` to this component.

Reply:
Done - i've also removed the `view` state and refactored the code to use the `LoadingProvider`